api/ceramic.yaml

```diff
@@ -632,3 +679,33 @@ components:
         name:
           type: string
           description: Name of the Ceramic network
+    StreamState:
+      title: State of a Ceramic stream
+      description: The state of a Ceramic stream as defined by the stream type aggregation and conflict resolution rules.
+      type: object
+      required:
+        - id
+        - event_cid
```

Comment:
also, should this be an array?  With multiprev we now support streams with multiple tips

Reply:
I don't think so, this API wants to assume there is always a canonical singular tip to a stream. What would the ceramic-sdk do if it gets back multiple tips?

Reply:
how do we expect to handle a CRDT type stream that can diverge and remerge?  What do you do when you have two different updates at the same log height?  You need a way to return both tips and then let the application construct a new event that merges the states and includes both tips in the prev array

Reply:
Long term yes and its possible to get multiple tips via the Flight SQL API. This API is for the narrow use case of the SDK needing to do a read in order to do a write. When we add CRDT type streams the SDK will not need to do a read before a write. It can just do the write. I imagine there are other similar differences with yet unknown stream types. As such I want this API to remain specific to the immediate needs.